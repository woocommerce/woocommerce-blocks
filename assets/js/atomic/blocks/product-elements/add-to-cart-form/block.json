{
	"name": "woocommerce/add-to-cart-form",
	"version": "1.0.0",
	"title": "Add to Cart with Options",
	"description": "Display a button so the customer can add a product to their cart. Options will also be displayed depending on product type. e.g. quantity, variation.",
	"attributes": {
		"isDescendentOfSingleProductBlock": {
			"type": "boolean",
			"default": false
		}
	},
	"category": "woocommerce",
	"keywords": [ "WooCommerce" ],
	"usesContext": ["postId"],
	"textdomain": "woo-gutenberg-products-block",
	"apiVersion": 2,
	"$schema": "https://schemas.wp.org/trunk/block.json"
}
