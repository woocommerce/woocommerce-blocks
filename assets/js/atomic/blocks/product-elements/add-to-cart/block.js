/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {
	AddToCartFormContextProvider,
	useAddToCartFormContext,
} from '@woocommerce/base-context';
import { useProductDataContext } from '@woocommerce/shared-context';
import { isEmpty } from 'lodash';
import { withProductDataContext } from '@woocommerce/shared-hocs';

/**
 * Internal dependencies
 */
import './style.scss';
import { AddToCartButton } from './shared';
import {
	SimpleProductForm,
	VariableProductForm,
	ExternalProductForm,
	GroupedProductForm,
} from './product-types';

/**
 * Product Add to Form Block Component.
 *
 * @param {Object}  props                    Incoming props.
 * @param {string}  [props.className]        CSS Class name for the component.
 * @param {boolean} [props.showFormElements] Should form elements be shown?
 * @return {*} The component.
 */
const Block = ( { className, showFormElements } ) => {
	const { product } = useProductDataContext();
	const componentClass = classnames(
		className,
		'wc-block-components-product-add-to-cart',
		{
			'wc-block-components-product-add-to-cart--placeholder': isEmpty(
				product
			),
		}
	);

	return (
		<AddToCartFormContextProvider
			product={ product }
			showFormElements={ showFormElements }
		>
			<div className={ componentClass }>
				<AddToCartForm />
			</div>
		</AddToCartFormContextProvider>
	);
};

/**
 * Renders the add to cart form using useAddToCartFormContext.
 */
const AddToCartForm = () => {
	const { showFormElements, productType } = useAddToCartFormContext();

	if ( showFormElements ) {
		if ( productType === 'variable' ) {
			return <VariableProductForm />;
		}
		if ( productType === 'grouped' ) {
			return <GroupedProductForm />;
		}
		if ( productType === 'external' ) {
			return <ExternalProductForm />;
		}
		if ( productType === 'simple' || productType === 'variation' ) {
			return <SimpleProductForm />;
		}
		return null;
	}

	return <AddToCartButton />;
};

Block.propTypes = {
	className: PropTypes.string,
};

export default withProductDataContext( Block );
