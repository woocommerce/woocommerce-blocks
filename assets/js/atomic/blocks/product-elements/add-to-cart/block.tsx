/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import classnames from 'classnames';
import {
	AddToCartFormContextProvider,
	useAddToCartFormContext,
} from '@woocommerce/base-context';
import { useProductDataContext } from '@woocommerce/shared-context';
import { isEmpty } from 'lodash';
import { withProductDataContext } from '@woocommerce/shared-hocs';

/**
 * Internal dependencies
 */
import './style.scss';
import { AddToCartButton } from './shared';
import {
	SimpleProductForm,
	VariableProductForm,
	ExternalProductForm,
	GroupedProductForm,
} from './product-types';

interface Props {
	/**
	 * CSS Class name for the component.
	 */
	className?: string;
	/**
	 * Whether or not to show form elements.
	 */
	showFormElements?: boolean;
}

/**
 * Renders the add to cart form using useAddToCartFormContext.
 */
const AddToCartForm = () => {
	const { showFormElements, productType } = useAddToCartFormContext();

	if ( showFormElements ) {
		if ( productType === 'variable' ) {
			return <VariableProductForm />;
		}
		if ( productType === 'grouped' ) {
			return <GroupedProductForm />;
		}
		if ( productType === 'external' ) {
			return <ExternalProductForm />;
		}
		if ( productType === 'simple' || productType === 'variation' ) {
			return <SimpleProductForm />;
		}
		return null;
	}

	return <AddToCartButton />;
};

/**
 * Product Add to Form Block Component.
 */
const Block = ( { className, showFormElements }: Props ) => {
	const { product } = useProductDataContext();
	const componentClass = classnames(
		className,
		'wc-block-components-product-add-to-cart',
		{
			'wc-block-components-product-add-to-cart--placeholder':
				isEmpty( product ),
		}
	);

	return (
		<AddToCartFormContextProvider
			product={ product }
			showFormElements={ showFormElements }
		>
			<div className={ componentClass }>
				<AddToCartForm />
			</div>
		</AddToCartFormContextProvider>
	);
};

Block.propTypes = {
	className: PropTypes.string,
};

export default withProductDataContext( Block );
