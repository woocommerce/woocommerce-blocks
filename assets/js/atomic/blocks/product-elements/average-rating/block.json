{
	"name": "woocommerce/product-average-rating",
	"version": "1.0.0",
	"title": "Product Average Rating (Beta)",
	"description": "Display the average rating of a product",
	"attributes": {
		"textAlign": {
			"type": "string"
		}
	},
	"category": "woocommerce",
	"keywords": [ "WooCommerce" ],
	"ancestor": [ "woocommerce/single-product" ],
	"textdomain": "woo-gutenberg-products-block",
	"apiVersion": 2,
	"$schema": "https://schemas.wp.org/trunk/block.json"
}
