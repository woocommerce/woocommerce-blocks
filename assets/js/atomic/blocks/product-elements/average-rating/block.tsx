/**
 * External dependencies
 */
import classnames from 'classnames';
import { useProductDataContext } from '@woocommerce/shared-context';
import { useStyleProps } from '@woocommerce/base-hooks';
import { __ } from '@wordpress/i18n';
import { withProductDataContext } from '@woocommerce/shared-hocs';

type ProductAverageRatingProps = {
	className?: string;
	textAlign?: string;
};

export const Block = ( props: ProductAverageRatingProps ): JSX.Element => {
	const { textAlign } = props;
	const styleProps = useStyleProps( props );
	const { product } = useProductDataContext();

	const className = classnames(
		styleProps.className,
		'wc-block-components-product-average-rating',
		{
			[ `has-text-align-${ textAlign }` ]: textAlign,
		}
	);

	return (
		<div className={ className } style={ styleProps.style }>
			{ Number( product.average_rating ) > 0
				? product.average_rating
				: __( 'No ratings', 'woo-gutenberg-products-block' ) }
		</div>
	);
};

export default withProductDataContext( Block );
