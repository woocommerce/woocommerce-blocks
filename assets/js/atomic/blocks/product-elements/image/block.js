/**
 * External dependencies
 */
import PropTypes from 'prop-types';
import { useState, Fragment } from '@wordpress/element';
import { __, sprintf } from '@wordpress/i18n';
import classnames from 'classnames';
import { PLACEHOLDER_IMG_SRC } from '@woocommerce/settings';
import {
	useInnerBlockLayoutContext,
	useProductDataContext,
} from '@woocommerce/shared-context';
import { withProductDataContext } from '@woocommerce/shared-hocs';
import { useStoreEvents } from '@woocommerce/base-context/hooks';

/**
 * Internal dependencies
 */
import ProductSaleBadge from './../sale-badge/block';
import './style.scss';

/**
 * Product Image Block Component.
 *
 * @param {Object} props                  Incoming props.
 * @param {string} [props.className]      CSS Class name for the component.
 * @param {string} [props.imageSizing]    Size of image to use.
 * @param {boolean} [props.showProductLink]   Whether or not to display a link to the product page.
 * @param {boolean} [props.showSaleBadge] Whether or not to display the on sale badge.
 * @param {string} [props.saleBadgeAlign] How should the sale badge be aligned if displayed.
 * @return {*} The component.
 */
export const Block = ( {
	className,
	imageSizing = 'full-size',
	showProductLink = true,
	showSaleBadge,
	saleBadgeAlign = 'right',
} ) => {
	const { parentClassName } = useInnerBlockLayoutContext();
	const { product } = useProductDataContext();
	const [ imageLoaded, setImageLoaded ] = useState( false );
	const { dispatchStoreEvent } = useStoreEvents();

	if ( ! product.id ) {
		return (
			<div
				className={ classnames(
					className,
					'wc-block-components-product-image',
					'wc-block-components-product-image--placeholder',
					{
						[ `${ parentClassName }__product-image` ]: parentClassName,
					}
				) }
			>
				<ImagePlaceholder />
			</div>
		);
	}
	const hasProductImages = !! product.images.length;
	const image = hasProductImages ? product.images[ 0 ] : null;
	const ParentComponent = showProductLink ? 'a' : Fragment;
	const anchorLabel = sprintf(
		/* translators: %s is referring to the product name */
		__( 'Link to %s', 'woo-gutenberg-products-block' ),
		product.name
	);
	const anchorProps = {
		href: product.permalink,
		rel: 'nofollow',
		...( ! hasProductImages && { 'aria-label': anchorLabel } ),
		onClick: () => {
			dispatchStoreEvent( 'product-view-link', {
				product,
			} );
		},
	};

	return (
		<div
			className={ classnames(
				className,
				'wc-block-components-product-image',
				{
					[ `${ parentClassName }__product-image` ]: parentClassName,
				}
			) }
		>
			<ParentComponent { ...( showProductLink && anchorProps ) }>
				{ !! showSaleBadge && (
					<ProductSaleBadge
						align={ saleBadgeAlign }
						product={ product }
					/>
				) }
				<Image
					fallbackAlt={ product.name }
					image={ image }
					onLoad={ () => setImageLoaded( true ) }
					loaded={ imageLoaded }
					showFullSize={ imageSizing !== 'cropped' }
				/>
			</ParentComponent>
		</div>
	);
};

const ImagePlaceholder = () => {
	return (
		<img src={ PLACEHOLDER_IMG_SRC } alt="" width={ 500 } height={ 500 } />
	);
};

const Image = ( { image, onLoad, loaded, showFullSize, fallbackAlt } ) => {
	const { thumbnail, src, srcset, sizes, alt } = image || {};
	const imageProps = {
		alt: alt || fallbackAlt,
		onLoad,
		hidden: ! loaded,
		src: thumbnail,
		...( showFullSize && { src, srcSet: srcset, sizes } ),
	};

	return (
		<>
			{ imageProps.src && (
				/* eslint-disable-next-line jsx-a11y/alt-text */
				<img data-testid="product-image" { ...imageProps } />
			) }
			{ ! loaded && <ImagePlaceholder /> }
		</>
	);
};

Block.propTypes = {
	className: PropTypes.string,
	fallbackAlt: PropTypes.string,
	showProductLink: PropTypes.bool,
	showSaleBadge: PropTypes.bool,
	saleBadgeAlign: PropTypes.string,
};

export default withProductDataContext( Block );
