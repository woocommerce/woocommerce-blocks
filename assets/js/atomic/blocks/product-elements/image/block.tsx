/**
 * External dependencies
 */
import { Fragment } from '@wordpress/element';
import { __, sprintf } from '@wordpress/i18n';
import classnames from 'classnames';
import { PLACEHOLDER_IMG_SRC } from '@woocommerce/settings';
import {
	useInnerBlockLayoutContext,
	useProductDataContext,
} from '@woocommerce/shared-context';
import { useStyleProps } from '@woocommerce/base-hooks';
import { withProductDataContext } from '@woocommerce/shared-hocs';
import { useStoreEvents } from '@woocommerce/base-context/hooks';
import type { HTMLAttributes } from 'react';

/**
 * Internal dependencies
 */
import ProductSaleBadge from '../sale-badge/block';
import './style.scss';
import { BlockAttributes, ImageSizing } from './types';

const ImagePlaceholder = (): JSX.Element => {
	return (
		<img
			src={ PLACEHOLDER_IMG_SRC }
			alt=""
			width={ undefined }
			height={ undefined }
		/>
	);
};

interface ImageProps {
	image?: null | {
		alt?: string | undefined;
		id: number;
		name: string;
		sizes?: string | undefined;
		src?: string | undefined;
		srcset?: string | undefined;
		thumbnail?: string | undefined;
	};
	loaded: boolean;
	showFullSize: boolean;
	fallbackAlt: string;
}

const Image = ( {
	image,
	loaded,
	showFullSize,
	fallbackAlt,
}: ImageProps ): JSX.Element => {
	const { thumbnail, src, srcset, sizes, alt } = image || {};
	const imageProps = {
		alt: alt || fallbackAlt,
		hidden: ! loaded,
		src: thumbnail,
		...( showFullSize && { src, srcSet: srcset, sizes } ),
	};

	return (
		<>
			{ imageProps.src && (
				/* eslint-disable-next-line jsx-a11y/alt-text */
				<img data-testid="product-image" { ...imageProps } />
			) }
			{ ! image && <ImagePlaceholder /> }
		</>
	);
};

type Props = BlockAttributes & HTMLAttributes< HTMLDivElement >;

export const Block = ( props: Props ): JSX.Element | null => {
	const {
		className,
		imageSizing = ImageSizing.SINGLE,
		showProductLink = true,
		showSaleBadge,
		saleBadgeAlign = 'right',
		...restProps
	} = props;
	const styleProps = useStyleProps( props );
	const { parentClassName } = useInnerBlockLayoutContext();
	const { product, isLoading } = useProductDataContext();
	const { dispatchStoreEvent } = useStoreEvents();

	if ( ! product.id ) {
		return (
			<div
				className={ classnames(
					className,
					'wc-block-components-product-image',
					{
						[ `${ parentClassName }__product-image` ]:
							parentClassName,
					},
					styleProps.className
				) }
				style={ styleProps.style }
			>
				<ImagePlaceholder />
			</div>
		);
	}
	const hasProductImages = !! product.images.length;
	const image = hasProductImages ? product.images[ 0 ] : null;
	const ParentComponent = showProductLink ? 'a' : Fragment;
	const anchorLabel = sprintf(
		/* translators: %s is referring to the product name */
		__( 'Link to %s', 'woo-gutenberg-products-block' ),
		product.name
	);
	const anchorProps = {
		href: product.permalink,
		...( ! hasProductImages && { 'aria-label': anchorLabel } ),
		onClick: () => {
			dispatchStoreEvent( 'product-view-link', {
				product,
			} );
		},
	};

	return (
		<div
			className={ classnames(
				className,
				'wc-block-components-product-image',
				{
					[ `${ parentClassName }__product-image` ]: parentClassName,
				},
				styleProps.className
			) }
			style={ styleProps.style }
		>
			<ParentComponent { ...( showProductLink && anchorProps ) }>
				{ !! showSaleBadge && (
					<ProductSaleBadge
						align={ saleBadgeAlign }
						{ ...restProps }
					/>
				) }
				<Image
					fallbackAlt={ product.name }
					image={ image }
					loaded={ ! isLoading }
					showFullSize={ imageSizing !== ImageSizing.THUMBNAIL }
				/>
			</ParentComponent>
		</div>
	);
};

export default withProductDataContext( Block );
