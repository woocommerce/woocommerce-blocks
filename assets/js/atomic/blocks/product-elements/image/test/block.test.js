/**
 * External dependencies
 */
import { render, screen, fireEvent } from '@testing-library/react';
import { ProductDataContextProvider } from '@woocommerce/shared-context';

/**
 * Internal dependencies
 */
import { Block } from '../block';

jest.mock( '@woocommerce/block-settings', () => ( {
	__esModule: true,
	PLACEHOLDER_IMG_SRC: 'placeholder.jpg',
} ) );

const productWithoutImages = {
	name: 'Test product',
	id: 1,
	fallbackAlt: 'Test product',
	permalink: 'http://test.com/product/test-product/',
	images: [],
};

const productWithImages = {
	name: 'Test product',
	id: 1,
	fallbackAlt: 'Test product',
	permalink: 'http://test.com/product/test-product/',
	images: [
		{
			id: 56,
			src: 'http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg',
			thumbnail:
				'http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg',
			srcset:
				'http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-768x767.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-416x415.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-100x100.jpg 100w',
			sizes: '(max-width: 800px) 100vw, 800px',
			name: 'logo-1.jpg',
			alt: '',
		},
		{
			id: 55,
			src:
				'http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1.jpg',
			thumbnail:
				'http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-324x324.jpg',
			srcset:
				'http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-768x768.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-416x416.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/beanie-with-logo-1-100x100.jpg 100w',
			sizes: '(max-width: 800px) 100vw, 800px',
			name: 'beanie-with-logo-1.jpg',
			alt: '',
		},
	],
};

describe( 'Product Image Block', () => {
	describe( 'with product link', () => {
		test( 'should render an anchor with the product image', () => {
			const component = render(
				<ProductDataContextProvider product={ productWithImages }>
					<Block productLink />
				</ProductDataContextProvider>
			);

			const image = screen.getByTestId( 'product-image' );
			fireEvent.load( image );

			expect( component ).toMatchSnapshot();
		} );
		test( 'should render an anchor with the placeholder image', () => {
			const component = render(
				<ProductDataContextProvider product={ productWithoutImages }>
					<Block productLink />
				</ProductDataContextProvider>
			);
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'without product link', () => {
		test( 'should render an anchor with the product image', () => {
			const component = render(
				<ProductDataContextProvider product={ productWithImages }>
					<Block productLink={ false } />
				</ProductDataContextProvider>
			);
			const image = screen.getByTestId( 'product-image' );
			fireEvent.load( image );
			expect( component ).toMatchSnapshot();
		} );
		test( 'should render an anchor with the placeholder image', () => {
			const component = render(
				<ProductDataContextProvider product={ productWithoutImages }>
					<Block productLink={ false } />
				</ProductDataContextProvider>
			);
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
