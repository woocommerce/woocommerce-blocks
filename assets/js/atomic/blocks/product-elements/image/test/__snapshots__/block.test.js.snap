// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image Block with product link should render an anchor with the placeholder image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wc-block-components-product-image"
      >
        <a
          aria-label="Link to Test product"
          href="http://test.com/product/test-product/"
          rel="nofollow"
        >
          <img
            alt=""
            height="500"
            src="placeholder.jpg"
            width="500"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wc-block-components-product-image"
    >
      <a
        aria-label="Link to Test product"
        href="http://test.com/product/test-product/"
        rel="nofollow"
      >
        <img
          alt=""
          height="500"
          src="placeholder.jpg"
          width="500"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product Image Block with product link should render an anchor with the product image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wc-block-components-product-image"
      >
        <a
          href="http://test.com/product/test-product/"
          rel="nofollow"
        >
          <img
            alt="Test product"
            data-testid="product-image"
            sizes="(max-width: 800px) 100vw, 800px"
            src="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg"
            srcset="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-768x767.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-416x415.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-100x100.jpg 100w"
          />
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wc-block-components-product-image"
    >
      <a
        href="http://test.com/product/test-product/"
        rel="nofollow"
      >
        <img
          alt="Test product"
          data-testid="product-image"
          sizes="(max-width: 800px) 100vw, 800px"
          src="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg"
          srcset="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-768x767.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-416x415.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-100x100.jpg 100w"
        />
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product Image Block without product link should render an anchor with the placeholder image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wc-block-components-product-image"
      >
        <img
          alt=""
          height="500"
          src="placeholder.jpg"
          width="500"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wc-block-components-product-image"
    >
      <img
        alt=""
        height="500"
        src="placeholder.jpg"
        width="500"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Product Image Block without product link should render an anchor with the product image 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wc-block-components-product-image"
      >
        <img
          alt="Test product"
          data-testid="product-image"
          sizes="(max-width: 800px) 100vw, 800px"
          src="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg"
          srcset="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-768x767.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-416x415.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-100x100.jpg 100w"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wc-block-components-product-image"
    >
      <img
        alt="Test product"
        data-testid="product-image"
        sizes="(max-width: 800px) 100vw, 800px"
        src="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg"
        srcset="http://localhost:10003/wp-content/uploads/2021/01/logo-1.jpg 800w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-300x300.jpg 300w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-150x150.jpg 150w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-768x767.jpg 768w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-324x324.jpg 324w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-416x415.jpg 416w, http://localhost:10003/wp-content/uploads/2021/01/logo-1-100x100.jpg 100w"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
